Sources/SpeziAccount/AccountNotifyStandard.swift

```diff
@@ -0,0 +1,18 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Spezi
+
+
+/// A `Spezi` Standard that allows to react to certain Account-based events.
+public protocol AccountNotifyStandard: Standard {
+    /// Notifies the Standard that the associated account was requested to be deleted by the user.
+    ///
+    /// Use this method to cleanup any account related data that might be associated with the account.
+    func deletedAccount() async
```

Comment:
Does it make sense to make this function throwing and actually call it before the account will be deleted so you have access to all credentials that might currently be associated with the account and only then ask the account service to remove anything credential-related?

Reply:
You are completely right. As cleanup actions typically involve network requests, it makes sense to have that throwing and (1) the account still associated if clean up fails and (2) have the account details still present for the cleanup action.

Addressed the feedback with the latest changes.

Reply:
Thank you, looks great 🚀